src/chart/generator/marker.cpp

```diff
@@ -135,31 +138,25 @@ void Marker::resetSize(bool horizontal)
 void Marker::setIdOffset(size_t offset)
 {
 	if (prevMainMarkerIdx.hasOneValue())
-		(*prevMainMarkerIdx).value += offset;
+		prevMainMarkerIdx->value += offset;
 	if (nextMainMarkerIdx.hasOneValue())
-		(*nextMainMarkerIdx).value += offset;
+		nextMainMarkerIdx->value += offset;
 	if (nextSubMarkerIdx.hasOneValue())
-		(*nextSubMarkerIdx).value += offset;
-}
-
-std::string Marker::toJSON() const
-{
-	std::string res;
-	appendToJSON(Conv::JSONObj{res});
-	return res;
+		nextSubMarkerIdx->value += offset;
 }
 
 Conv::JSONObj &&Marker::appendToJSON(Conv::JSONObj &&jsonObj) const
 {
-	return std::move(jsonObj)("categories",
-	    cellInfo.categories)("values", cellInfo.values)("index", idx);
+	if (cellInfo) return std::move(jsonObj).merge(cellInfo->json);
+	return std::move(jsonObj);
```

Comment:
Check which cases when we run here

Reply:
@simzer This is a behavior change. Some markers are disabled by both plots (source/target) but should be drawn because of a connection (previous or next enabled marker) on a line/area plot. For these markers, the cellInfo is not generated. This means disabled markers' info is not sent on a 'plot-marker-draw' event.

Reply:
(note: if the marker "needs" a tooltip, the cellInfo is generated)

Reply:
I think it's ok, this whole part will be changed with new marker morph.